ios/Plugin/CapacitorUpdater.swift

```diff
@@ -555,75 +556,202 @@ extension CustomError: LocalizedError {
         UserDefaults.standard.synchronize()
         print("\(self.TAG) Current bundle set to: \((bundle ).isEmpty ? BundleInfo.ID_BUILTIN : bundle)")
     }
+    private func getChecksum(filePath: URL) -> String {
+        let bufferSize = 1024 * 1024 * 5 // 5 MB
+        var checksum = uLong(0)
 
-    public func download(url: URL, version: String, sessionKey: String) throws -> BundleInfo {
-        let semaphore: DispatchSemaphore = DispatchSemaphore(value: 0)
-        let id: String = self.randomString(length: 10)
-        var checksum: String = ""
+        do {
+            let fileHandle = try FileHandle(forReadingFrom: filePath)
+            defer {
+                fileHandle.closeFile()
+            }
 
-        var mainError: NSError?
-        let destination: DownloadRequest.Destination = { _, _ in
-            let documentsURL: URL = FileManager.default.urls(for: .documentDirectory, in: .userDomainMask)[0]
-            let fileURL: URL = documentsURL.appendingPathComponent(self.randomString(length: 10))
+            while autoreleasepool(invoking: {
+                let fileData = fileHandle.readData(ofLength: bufferSize)
+                if fileData.count > 0 {
+                    checksum = fileData.withUnsafeBytes {
+                        crc32(checksum, $0.bindMemory(to: Bytef.self).baseAddress, uInt(fileData.count))
+                    }
+                    return true // Continue
+                } else {
+                    return false // End of file
+                }
+            }) {}
 
-            return (fileURL, [.removePreviousFile, .createIntermediateDirectories])
+            return String(format: "%08X", checksum).lowercased()
+        } catch {
+            print("\(self.TAG) Cannot get checksum: \(filePath.path)", error)
+            return ""
         }
-        let request = AF.download(url, to: destination)
+    }
+    //Do a GET request on the url in order to extract the Content-Length header, ask only for the 10 first bytes in order to not download the full content.
+    func fetchFileSize(url: URL) -> Int? {
```

Comment:
This function is not required.

You can get the length of the file that you are downloading by doing the following

```swift
        //Opening connection for streaming the bytes
        AF.streamRequest(url, headers: requestHeaders).validate().onHTTPResponse(perform: { response in
            print("magic \(response.headers)")
        }).responseStream { [weak self] streamResponse in
```

And it will return the following:

```
magic Etag: "1869b2563b642e042abdf422cf58884b"
Strict-Transport-Security: max-age=31536000; includeSubDomains
X-Amz-Request-Id: 17EB9AD5E895213C
X-Amz-Id-2: dd9025bab4ad464b049177c95eb6ebf374d3b3fd1af9251148b658df7ac2e3e8
X-Content-Type-Options: nosniff
Content-Range: bytes 0-100664/100665
Date: Wed, 14 Aug 2024 13:18:45 GMT
Content-Type: binary/octet-stream
Server: MinIO
Vary: Origin, Accept-Encoding
Last-Modified: Wed, 14 Aug 2024 12:44:58 GMT
Accept-Ranges: bytes
X-Xss-Protection: 1; mode=block
Content-Length: 100665 <--------- IMPORTANT HEADER
```

Reply:
Thanks for the help ! I adressed a solution and removed `fetchFileSize` in [fbb44b4](https://github.com/Cap-go/capacitor-updater/pull/400/commits/fbb44b4b41cffb22b41e15db902f4b25c8531e91)